lib/src/services/jira_api.ts

```diff
@@ -570,48 +585,162 @@ const jiraClient = (
             ],
             expand: ["changelog"],
           }),
+
+        //   const fs = require("fs");
+        //   const data = JSON.stringify(
+        //     reporter.report(T.array(GenericJiraIssue).decode(result["issues"])),
+        //     null,
+        //     4
+        //   );
+        //   try {
+        //     fs.writeFileSync("issues_cloud.json", data);
+        //     console.log("JSON data is saved.");
+        //   } catch (error) {
+        //     console.error(error);
+        //   }
+
+        //   return result;
+        // },
+
         (error: unknown) =>
           `Error fetching details from jira with query [${jql}] - [${JSON.stringify(
             error
           )}].`
       );
 
-      const parsed = (
+      // const parsed = (
+      //   response: JiraApi.JsonResponse
+      // ): TE.TaskEither<string, ReadonlyArray<GenericJiraIssue>> =>
+      //   TE.fromEither(
+      //     decode(
+      //       "issues",
+      //       response["issues"],
+      //       // eslint-disable-next-line @typescript-eslint/unbound-method
+      //       T.readonly(T.array(GenericJiraIssue)).decode
+      //     )
+      //   );
+
+      // const isCloud = jiraHost.includes("atlassian") ? true : false; //parseCloudJira(response) : parseOnPremJira(response);
+
+      const convertIssueType = (
+        response: JiraApi.JsonResponse
+      ): TE.TaskEither<string, readonly GenericJiraIssue[]> => {
+        return jiraHost.includes("atlassian")
+          ? pipe(
+              parseCloudJira(response), //reponse to cloudIssueType
+              TE.chain(
+                TE.traverseSeqArray((cloudJiraIssue) =>
+                  cloudJiraToGeneric(cloudJiraIssue)
+                )
+              )
+            )
+          : pipe(
+              parseOnPremJira(response),
+              TE.chain(
+                TE.traverseSeqArray((onPremJiraIssue) =>
+                  onPremJiraToGeneric(onPremJiraIssue)
+                )
+              )
+            );
+      };
+
+      const parseOnPremJira = (
```

Comment:
These are the same except for the decoder used?

Reply:
There is a minor difference between `OnPremJiraIssue` and `CloudJiraIssue`, which means that they have to be decoded seperately into a generic Jira issue. This [comment](https://agiledigital.atlassian.net/browse/IE-3?focusedCommentId=15010) by Yoonji highlights it.

Reply:
resolved